rosbag2_transport/include/rosbag2_transport/record_options.hpp

```diff
@@ -32,6 +32,7 @@ struct RecordOptions
   std::string node_prefix = "";
   std::string compression_mode = "";
   std::string compression_format = "";
+  std::string qos_profiles = "";
```

Comment:
Can we try to have the API provide something more structured? Maybe `std::unordered_map<std::string, rclcpp::QoS> topic_qos_profile_overrides;` Since this is a public API and we may have external code writing calls against it, this is probably too error prone.

Worst case, let's name this something a little more descriptive, e.g. `yaml_serialized_per_topic_qos_profile_overrides_map`. But as a last resort.